### PR TITLE
Globals: Add a Loading spinner next to the globals popover toggle button

### PR DESCRIPTION
**Pull request by udidol**
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Globals: Add a Loading spinner next to the globals popover toggle button to indicate when the list of globals has loaded

## Quality assurance

- [X] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: core/kits/assets/js/globals/global-select-behavior.js [hunk 1]
```diff
@@ -233,14 +233,17 @@ export default class GlobalControlSelect extends Marionette.Behavior {
 
 	printGlobalToggleButton() {
 		const $globalToggleButton = jQuery( '<div>', { class: this.getClassNames().popoverToggle + ' elementor-control-unit-1' } ),
-			$globalPopoverToggleIcon = jQuery( '<i>', { class: 'eicon-globe' } );
+			$globalPopoverToggleIcon = jQuery( '<i>', { class: 'eicon-globe' } ),
+			$globalsLoadingSpinner = jQuery( '<span>', { class: 'elementor-control-spinner' } )
+				.html( '<i class="eicon-spinner eicon-animation-spin"></i>&nbsp;</span>' );
```
Comment (matipojo) [inline core/kits/assets/js/globals/global-select-behavior.js:238]:
```suggestion
				.html( '<i class="eicon-spinner eicon-animation-spin"></i></span>' );
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/kits/assets/js/globals/global-select-behavior.js (hunk 1 above)
Comment (udidol) [reply core/kits/assets/js/globals/global-select-behavior.js]: @matipojo I replaced the space with a CSS margin. The space character actually gave a margin from the globals toggle.

File: core/kits/assets/js/globals/global-select-behavior.js [hunk 2]
```diff
@@ -352,11 +361,15 @@ export default class GlobalControlSelect extends Marionette.Behavior {
 
 		globalMeta.title = this.ui.globalNameInput.val();
 
+		this.ui.$globalsLoadingSpinner.show();
```
Comment (kobizz) [inline core/kits/assets/js/globals/global-select-behavior.js:364]: Move it inside `createNewGlobal` method

File: core/kits/assets/js/globals/global-select-behavior.js [hunk 3]
```diff
@@ -352,11 +361,15 @@ export default class GlobalControlSelect extends Marionette.Behavior {
...
+
 		this.createNewGlobal( globalMeta )
 			.then( ( result ) => {
 				const $globalPreview = this.view.createGlobalItemMarkup( result.data );
 
 				this.ui.globalPreviewsContainer.append( $globalPreview );
+
+				this.ui.$globalsLoadingSpinner.hide();
```
Comment (kobizz) [inline core/kits/assets/js/globals/global-select-behavior.js:372]: Move all `then` functionality to the internal `createNewGlobal` `then`

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by kobizz — APPROVED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**